habitat-lab/habitat/articulated_agent_controllers/humanoid_base_controller.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import magnum as mn
+import numpy as np
+
+
+class Pose:
+    def __init__(self, joints_quat, root_transform):
+        """
+        Contains a single humanoid pose
+            :param joints_quat: list or array of num_joints * 4 elements, with the rotation quaternions
+            :param root_transform: Matrix4 with the root trasnform.
+        """
+        self.joints = list(joints_quat)
+        self.root_transform = root_transform
+
+
+class Motion:
+    """
+    Contains a sequential motion, corresponding to a sequence of poses
+        :param joints_quat_array: num_poses x num_joints x 4 array, containing the join orientations
```

Comment:
Are we assuming that all humanoids will have quaternions only for joint rep? This may apply to AMASS, but is not generic.

The [ManagedBulletArticulatedObject](https://aihabitat.org/docs/habitat-sim/habitat_sim.physics.ManagedBulletArticulatedObject.html) class has functionality to represent and query degrees of freedom (dofs) generically.